Cannot use Endpoint.of as a parameter decorator

The following code results in `Error: src\app.ts(18,15): error TS1239: Unable to resolve signature of parameter decorator when called as an expression.
  Cannot invoke an expression whose type lacks a call signature. Type 'Endpoint' has no compatible call signatures.
`

I'd like to resolve my endpoints in the constructor while also using `autoinject` in typescript. Is this possible? The example on Aurelia Hub for Resolves in TS mode seems to suggest it: http://aurelia.io/hub.html#/doc/article/aurelia/dependency-injection/latest/dependency-injection-basics/6

```
@autoinject
export class App {
  constructor(private authService: AuthService,
              private eventAggregator: EventAggregator,
              @Endpoint.of('protected-api')
              private privateResource: Endpoint){}
}
```

--- a/doc/usage.md
+++ b/doc/usage.md
@@ -38,4 +38,14 @@ export class MyClass {
 }
 ```
 
-Note: Resolvers cannot be used with @autoinject. See [http://aurelia.io/docs/fundamentals/dependency-injection#resolvers]
+To use Resolvers with @autoinject use @inject as parameter decorator
+```js
+import {inject, autoinject} from 'aurelia-framework';
+import {Endpoint, Rest} from 'aurelia-api';
+
+@autoinject()
+export class MyClass {
+  constructor(public @inject(Endpoint.of('api')) apiEndpoint: Rest) {
+  }
+}
+```

